kitsune/community/utils.py

```diff
@@ -1,75 +1,116 @@
 from datetime import datetime, date, timedelta
+from django.core.cache import cache
 from django.conf import settings
+from django.db.models import Count
 
 from kitsune.customercare.models import ReplyMetricsMappingType
 from kitsune.products.models import Product
 from kitsune.questions.models import AnswerMetricsMappingType
 from kitsune.search.es_utils import F
-from kitsune.users.models import UserMappingType
+from kitsune.users.models import User
+from kitsune.users.templatetags.jinja_helpers import profile_avatar
 from kitsune.wiki.models import RevisionMetricsMappingType
 
 
 # This should be the higher than the max number of contributors for a
 # section.  There isn't a way to tell ES to just return everything.
 BIG_NUMBER = 3000
 
+# This should be higher than the max number of questions, revisions or tweets
+HUGE_NUMBER = 1500000
+
 
 def top_contributors_questions(start=None, end=None, locale=None, product=None,
-                               count=10, page=1):
+                               count=10, page=1, use_cache=True):
     """Get the top Support Forum contributors."""
     # Get the user ids and contribution count of the top contributors.
-    query = (
-        AnswerMetricsMappingType
-        .search()
-        .facet('creator_id', filtered=True, size=BIG_NUMBER))
+
+    cache_key = 'top_contributors_questions_{}_{}_{}_{}_{}_{}'.format(start, end, locale,
```

Comment:
There should be unicode data. There is a test fail because of that!
```python
ERROR: test_product_filter (kitsune.questions.tests.test_templates.QuestionsTemplateTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/app/kitsune/questions/tests/test_templates.py", line 1146, in test_product_filter
    check(p1.slug, [q2])
  File "/app/kitsune/questions/tests/test_templates.py", line 1132, in check
    response = self.client.get(url)
  File "/usr/local/lib/python2.7/dist-packages/django/test/client.py", line 500, in get
    **extra)
  File "/usr/local/lib/python2.7/dist-packages/django/test/client.py", line 303, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/usr/local/lib/python2.7/dist-packages/django/test/client.py", line 379, in generic
    return self.request(**r)
  File "/app/kitsune/sumo/tests/__init__.py", line 116, in request
    return super(LocalizingClient, self).request(**request)
  File "/usr/local/lib/python2.7/dist-packages/django/test/client.py", line 466, in request
    six.reraise(*exc_info)
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/mobility/decorators.py", line 23, in wrapper
    return f(request, *args, **kw)
  File "/app/kitsune/questions/views.py", line 301, in question_list
    locale=forum_locale, product=product)
  File "/app/kitsune/community/utils.py", line 30, in top_contributors_questions
    cached = cache.get(cache_key, None)
  File "/usr/local/lib/python2.7/dist-packages/django/core/cache/backends/memcached.py", line 83, in get
    key = self.make_key(key, version=version)
  File "/usr/local/lib/python2.7/dist-packages/django/core/cache/backends/memcached.py", line 76, in make_key
    return force_str(super(BaseMemcachedCache, self).make_key(key, version))
  File "/usr/local/lib/python2.7/dist-packages/django/core/cache/backends/base.py", line 102, in make_key
    new_key = self.key_func(key, self.key_prefix, version)
  File "/usr/local/lib/python2.7/dist-packages/django/core/cache/backends/base.py", line 35, in default_key_func
    return '%s:%s:%s' % (key_prefix, version, key)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc4 in position 43: ordinal not in range(128)
```